test/EFCore.MySql.FunctionalTests/MigrationSqlGeneratorMySqlTest.cs

```diff
@@ -151,6 +153,45 @@ public virtual void CreateTableUlongAutoincrement()
                 Sql);
         }
 
+        [Theory]
+        [InlineData(true, false, CharSetAppendBehavior.AppendToAllAnsiColumns, CharSet.Latin1, CharSet.Utf8mb4, "")]
+        [InlineData(true, false, CharSetAppendBehavior.AppendToUnicodeIndexAndKeyColumns, CharSet.Latin1, CharSet.Utf8mb4, "")]
+        [InlineData(true, false, CharSetAppendBehavior.AppendToAllUnicodeColumns, CharSet.Latin1, CharSet.Utf8mb4, "utf8mb4")]
+        [InlineData(true, false, CharSetAppendBehavior.AppendToAllColumns, CharSet.Latin1, CharSet.Utf8mb4, "utf8mb4")]
+        [InlineData(true, false, CharSetAppendBehavior.AppendToAllColumns, CharSet.Latin1, CharSet.Ucs2, "ucs2")]
+        [InlineData(false, false, CharSetAppendBehavior.AppendToAllUnicodeColumns, CharSet.Latin1, CharSet.Utf8mb4, "")]
+        [InlineData(false, false, CharSetAppendBehavior.AppendToAnsiIndexAndKeyColumns, CharSet.Latin1, CharSet.Utf8mb4, "")]
+        [InlineData(false, false, CharSetAppendBehavior.AppendToAllAnsiColumns, CharSet.Latin1, CharSet.Utf8mb4, "latin1")]
+        [InlineData(false, false, CharSetAppendBehavior.AppendToAllColumns, CharSet.Latin1, CharSet.Utf8mb4, "latin1")]
+        [InlineData(false, false, CharSetAppendBehavior.AppendToAllColumns, CharSet.Ucs2, CharSet.Utf8mb4, "ucs2")]
+        public virtual void AddColumnOperation_with_charset(bool isUnicode, bool isIndex, CharSetAppendBehavior charSetAppendBehavior,
+            CharSet ansiCharSet, CharSet unicodeCharSet, string expectedCharSet)
+        {
+            Generate(
+                modelBuilder => modelBuilder.Entity("Person").Property<string>("Name").IsUnicode(isUnicode),
```

Comment:
@AndriySvyryd how can I add `.IsIndex()` here so that the "Is Index" codepath can be tested?

Reply:
Something like
```C#
Generate(
    modelBuilder => modelBuilder.Entity(
        "Person",
        eb =>
        {
            eb.Property<string>("Name").IsUnicode(isUnicode);
            if (isIndex)
            {
                eb.HasIndex("Name");
            }
        }),
```

Reply:
Thanks I'll add that.